modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -654,6 +663,10 @@ XContentType getXContentType() {
         return documentXContentType;
     }
 
+    public String getQueryName() {
```

Comment:
There is a getter in `AbstractQueryBuilder#queryName` so no need to add a new function.

Reply:
I noticed that getter but it returns the `protected String queryName` field in the `AbstractQueryBuilder`, i need to test for the `private String name` field in the `PercolateQueryBuilder`

Reply:
hi @jimczi as i stated in the previous comment `AbstractQueryBuilder#queryName` is a getter of another field that's why i added the `getQueryName`

do you need me to update any other thing to move this pr forward?

Reply:
Right, sorry for the confusion. So let's change the name to `getName` (for consistency with `setName`) and add javadocs ?

Reply:
@jimczi changing name to `getName` will overwrite the `AbstractQueryBuilder.getName` which returns another field (`PercolateQueryBuilder` => `public static final String NAME`)

Reply:
ok got it now, sorry for being slow and thanks for explaining.